lib/class-wp-rest-menu-items-controller.php

```diff
@@ -474,66 +571,13 @@ protected function prepare_item_for_database( $request ) {
 			}
 		}
 
-		foreach ( array( 'menu-item-object-id', 'menu-item-parent-id' ) as $key ) {
-			// Note we need to allow negative-integer IDs for previewed objects not inserted yet.
-			$prepared_nav_item[ $key ] = intval( $prepared_nav_item[ $key ] );
-		}
-
-		foreach ( array( 'menu-item-type', 'menu-item-object', 'menu-item-target' ) as $key ) {
-			$prepared_nav_item[ $key ] = sanitize_key( $prepared_nav_item[ $key ] );
-		}
-
-		// Valid xfn and classes are an array.
-		foreach ( array( 'menu-item-xfn', 'menu-item-classes' ) as $key ) {
-			$value = $prepared_nav_item[ $key ];
-			if ( ! is_array( $value ) ) {
-				$value = wp_parse_list( $value );
-			}
-			$prepared_nav_item[ $key ] = implode( ' ', array_map( 'sanitize_html_class', $value ) );
-		}
-
-		// Apply the same filters as when calling wp_insert_post().
-
-		/** This filter is documented in wp-includes/post.php */
-		$prepared_nav_item['menu-item-title'] = wp_unslash( apply_filters( 'title_save_pre', wp_slash( $prepared_nav_item['menu-item-title'] ) ) );
-
-		/** This filter is documented in wp-includes/post.php */
-		$prepared_nav_item['menu-item-attr-title'] = wp_unslash( apply_filters( 'excerpt_save_pre', wp_slash( $prepared_nav_item['menu-item-attr-title'] ) ) );
-
-		/** This filter is documented in wp-includes/post.php */
-		$prepared_nav_item['menu-item-description'] = wp_unslash( apply_filters( 'content_save_pre', wp_slash( $prepared_nav_item['menu-item-description'] ) ) );
-
-		// Valid url.
-		if ( '' !== $prepared_nav_item['menu-item-url'] ) {
-			$prepared_nav_item['menu-item-url'] = esc_url_raw( $prepared_nav_item['menu-item-url'] );
-			if ( '' === $prepared_nav_item['menu-item-url'] ) {
-				// Fail sanitization if URL is invalid.
-				return new WP_Error( 'invalid_url', __( 'Invalid URL.', 'gutenberg' ), array( 'status' => 400 ) );
-			}
-		}
-		// Only draft / publish are valid post status for menu items.
-		if ( 'publish' !== $prepared_nav_item['menu-item-status'] ) {
-			$prepared_nav_item['menu-item-status'] = 'draft';
-		}
-
-		$prepared_nav_item = (object) $prepared_nav_item;
-
-		/**
-		 * Filters a post before it is inserted via the REST API.
-		 *
-		 * The dynamic portion of the hook name, `$this->post_type`, refers to the post type slug.
-		 *
-		 * @param stdClass        $prepared_post An object representing a single post prepared
-		 *                                       for inserting or updating the database.
-		 * @param WP_REST_Request $request       Request object.
-		 */
-		return apply_filters( "rest_pre_insert_{$this->post_type}", $prepared_nav_item, $request );
+		$request['prepared_nav_item'] = $prepared_nav_item;
```

Comment:
I'm really not a fan of us passing data thru the request object like this. What does it look like to stick with a `prepare_item_for_database` method that does this with already validated data.

Reply:
Could not agree more just put the return and filter back. This change is completely unnessary

Reply:
This line seems to completely overwrite the existing request. Please remove as it breaks things.